builtin/credential/aws-ec2/path_role.go

```diff
@@ -271,6 +291,10 @@ func (b *backend) pathRoleCreateUpdate(
 		roleEntry.MaxTTL = time.Duration(data.Get("max_ttl").(int)) * time.Second
 	}
 
+	if roleEntry.MaxTTL < roleEntry.TTL {
```

Comment:
Should maybe guard this (if it isn't above, hard to tell from the diff) against this being set explicitly (or implicitly) to zero -- it should be valid for the set max TTL to be 0 while the normal TTL is set.